os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -133,6 +133,20 @@ void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size
 	for (int idx = startidx; idx <= endidx; idx++) {
 		heapinfo_parse_heap(&heap[idx], HEAPINFO_DETAIL_ALL, HEAPINFO_PID_ALL);
 	}
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	struct mm_heap_s *kheap = kmm_get_baseheap();
+	if (heap != kheap) {
+		mfdbg("#####################################################################################\n\n");
+		mfdbg("*************************************************************************************\n");
+		mfdbg("                          Summary of Kernel heap memory usage                        \n");
+		mfdbg("*************************************************************************************\n\n");
+		for (int idx = HEAP_START_IDX; idx <= HEAP_END_IDX; idx++) {
+			heapinfo_parse_heap(&kheap[idx], HEAPINFO_SIMPLE, HEAPINFO_PID_ALL);
+		}
+		mfdbg("#####################################################################################\n\n");
```

Comment:
Here, you are printing kernel heap details. 
But there might be cases where we have more than one app heap (suppose we are running 2 apps).
In this case you need to print details of the other app also. 
For this, please print the details of all heaps in g_app_heap_table, excluding the current app.